src/lib/getClasses.ts

```diff
@@ -0,0 +1,49 @@
+import * as _ from 'lodash'
+import { combineRules } from 'fela'
+import callable from './callable'
+
+import {
+  ComponentPartStyleFunction,
+  ComponentStyleFunctionParam,
+  IComponentPartClasses,
+  IComponentPartStylesInput,
+  IRenderer,
+  OneOrArray,
+} from '../../types/theme'
+import { toCompactArray } from './index'
+
+/**
+ * Returns a string of HTML classes.
+ * Renders one or many component styles (objects of component parts) to the DOM.
+ */
+const getClasses = (
+  renderer: IRenderer,
+  componentStyles: OneOrArray<IComponentPartStylesInput>,
+  styleParam: ComponentStyleFunctionParam,
+): IComponentPartClasses => {
+  const stylesArr = toCompactArray(componentStyles)
+
+  // root, icon, etc.
+  const componentParts: string[] = stylesArr.reduce((acc, next) => {
+    return next ? _.union(acc, _.keys(next)) : acc
```

Comment:
`next` should always be truthy because of the above call to `toCompactArray`, if I'm understanding this correctly.